utils/kubernetes/crd.go

```diff
@@ -35,7 +34,7 @@ func GetAllCustomResourcesInCluster(ctx context.Context, client rest.Interface)
 	}
 	var xcrd CRD
 	gvks := []*schema.GroupVersionResource{}
-	err = encoding.Unmarshal(crdresult, &xcrd)
```

Comment:
`encoding.Unmarshal` is not deprecated, in fact recommended to be used. 
It is the encoder recently added in Meshkit, which supports both `json/yaml`.